File: mvn-advent-of-code/src/test/java/advent/twenty_seventeen/Day01InverseCaptchaTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import advent.twenty_sixteen.Day01NoTimeForATaxicab;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day01InverseCaptchaTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day01InverseCaptcha("/twenty_seventeen/Day01-InverseCaptcha-input.txt");

        assertEquals(1253, dp.getPart1Answer().intValue());
        assertEquals(1278, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day01InverseCaptcha("/twenty_seventeen/Day01-InverseCaptcha-example#1.txt");

        assertEquals(16, dp.getPart1Answer().intValue());
        assertEquals(10, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample2() {
        DailyProblem<Integer, Integer> dp = new Day01InverseCaptcha("/twenty_seventeen/Day01-InverseCaptcha-example#2.txt");

        assertEquals(3, dp.getPart1Answer().intValue());
        assertEquals(26, dp.getPart2Answer().intValue());
    }
}

File: kotlin-advent-of-code/src/test/kotlin/advent/twenty_seventeen/Day01InverseCaptchaTest.kt
package advent.twenty_seventeen

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class Day01InverseCaptchaTest {
    @Test
    fun checkReal() {
        val dp = Day01InverseCaptcha("/twenty_seventeen/Day01-InverseCaptcha-input.txt")

        assertEquals(1253, dp.part1Answer)
        assertEquals(1278, dp.part2Answer)
    }

    @Test
    fun checkExample1() {
        val dp = Day01InverseCaptcha("/twenty_seventeen/Day01-InverseCaptcha-example#1.txt")

        assertEquals(16, dp.part1Answer)
        assertEquals(10, dp.part2Answer)
    }

    @Test
    fun checkExample2() {
        val dp = Day01InverseCaptcha("/twenty_seventeen/Day01-InverseCaptcha-example#2.txt")

        assertEquals(3, dp.part1Answer)
        assertEquals(26, dp.part2Answer)
    }
}